### PR TITLE
Uses new dictclass types

### PR DESCRIPTION
**Pull request by mikesol**
This exposed several hidden bugs that were untested but only revealed through stronger typing.

### CODE REVIEW

File: tests/server/server/test_callbacks.py [hunk 1]
```diff
@@ -29,107 +31,112 @@ def test_json():
...
-    assert 'value' == new_response['headers']['X-Echo-Header']
+    assert 'Hello' == new_response.bodyAsJson['message']
+    assert 'value' == new_response.headers['X-Echo-Header']
 
 def test_text_full():
     callback_manager.load('tests/server/server/callbacks')
     storage_manager.clear()
 
-    request = Request(method='post',
+    request = RequestBuilder.from_dict(dict(method='post',
```
Comment (ksaaskil) [inline tests/server/server/test_callbacks.py]: Now that `Request` etc. are proper classes, we shouldn't need "builder" and "writer" classes but could use instance or static methods in `Request` class directly I think.

Comment (mikesol) [reply tests/server/server/test_callbacks.py]: Definitely possible. The issue is that builder has some additional logic to transform fields, etc. But perhaps this should be offloaded to a constructor. Unless we want to keep the dataclass "pure" (no logic, just data).

**Comment by mikesol** [comment]
> It's a bit big PR and I don't anymore understand that well what's going on but looks good! 👍

This is a big problem with the PR. As it changes the typing system entirely, almost every file with types is touched, which is almost every file. I agree we should avoid PRs like this.

File: tests/server/server/test_callbacks.py (hunk 1 above)
Comment (ksaaskil) [reply tests/server/server/test_callbacks.py]: Yep, `Request.from_dict` just reads a lot better than `RequestBuilder.from_dict`. We could also make the dataclasses frozen to clearly indicate they're not supposed to be mutated.

**Pull request merged**